src/utils.c

```diff
@@ -53,22 +53,62 @@ int local_strchr(char *string, char ch) {
     return -1;
 }
 
-uint32_t getV(txContent_t *txContent) {
-    uint32_t v = 0;
-    if (txContent->vLength == 1) {
-        v = txContent->v[0];
-    } else if (txContent->vLength == 2) {
-        v = (txContent->v[0] << 8) | txContent->v[1];
-    } else if (txContent->vLength == 3) {
-        v = (txContent->v[0] << 16) | (txContent->v[1] << 8) | txContent->v[2];
-    } else if (txContent->vLength == 4) {
-        v = (txContent->v[0] << 24) | (txContent->v[1] << 16) | (txContent->v[2] << 8) |
-            txContent->v[3];
-    } else if (txContent->vLength != 0) {
-        PRINTF("Unexpected v format\n");
+// Almost like U4BE except that it takes `size` as a parameter.
+// The `strict` parameter defines whether we should throw in case of a length > 4.
+uint32_t u32_from_BE(uint8_t *in, uint8_t size, bool strict) {
+    uint32_t res = 0;
+    if (size == 1) {
+        res = in[0];
+    } else if (size == 2) {
+        res = (in[0] << 8) | in[1];
+    } else if (size == 3) {
+        res = (in[0] << 16) | (in[1] << 8) | in[2];
+    } else if (size == 4) {
+        res = (in[0] << 24) | (in[1] << 16) | (in[2] << 8) | in[3];
+    } else if (strict && size != 0) {
+        PRINTF("Unexpected format\n");
         THROW(EXCEPTION);
     }
-    return v;
+    return res;
+}
+
+// Converts a uint32_t to a string.
+void u32_to_str(char *dest, uint32_t in, uint8_t dest_size) {
```

Comment:
actually you can use `snprintf` to do this :f:
`snprintf(dest, dest_size, "%d ", in);`